internal/provider/auth.go

```diff
@@ -143,15 +138,15 @@ func (l *AuthLoginCommon) Init(d *schema.ResourceData, authField string, validat
 		return err
 	}
 
-	l.mount = path
-	l.params = params
-
 	for _, vf := range validators {
-		if err := vf(d); err != nil {
+		if err := vf(d, params); err != nil {
```

Comment:
Updating the signature for `validateFunc` and passing in the params here allows us to maintain existing behavior for `l.params=nil` returned when validation fails. This way we don't have to update any existing tests, since we only update `l.params` once all validators have passed